### PR TITLE
[#3209]test(web): doris web e2e test

### PR DESCRIPTION
**Pull request by LauraXia123**
### What changes were proposed in this pull request?

Add test create doris catalog to the web e2e test

### Why are the changes needed?

Fix: #3209

### Does this PR introduce _any_ user-facing change?
N/A

### How was this patch tested?
manual

### CODE REVIEW

**Comment by yuqi1129** [comment]
@LauraXia123 
I have solved the problem related to Doris, there still some issues about fornt-end test, please take a look.

**Comment by LauraXia123** [comment]
@yuqi1129 Thanks a lot. I'll check the left issues.

**Comment by LauraXia123** [comment]
All the test case have passed, but I can't start the doris container locally, can you try it locally for me? @xunliu

**Comment by zhoukangcn** [comment]
> All the test case have passed, but I can't start the doris container locally, can you try it locally for me? @xunliu

Did you start doris successfully? If you still have problem, I can help

**Comment by LauraXia123** [comment]
image
image
@zhoukangcn  You can take a look my stack information above.

**Comment by zhoukangcn** [comment]
@LauraXia123 you can exec command `docker images|grep doris` to make sure doris Image exist (`datastrato/gravitino-ci-doris:0.1.3`)

If Doris Image not exist, you can use command `docker pull datastrato/gravitino-ci-doris:0.1.3` before CI

If Doris Image exist, you can see log in `catalogs/catalog-jdbc-doris/build` , include `testcontainers.log`, `catalog-jdbc-doris-integration-test.log` and send it to me

**Comment by LauraXia123** [comment]
The image is exist but not log files.
image
image

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+public class CatalogsPageDorisTest extends AbstractWebIT {
+  MetalakePage metalakePage = new MetalakePage();
+  CatalogsPage catalogsPage = new CatalogsPage();
+
+  private static final ContainerSuite containerSuite = ContainerSuite.getInstance();
+  protected static GravitinoAdminClient gravitinoClient;
+  private static GravitinoMetalake metalake;
+
+  protected static String gravitinoUri = "http://127.0.0.1:8090";
+  protected static String dorisUri = "http://127.0.0.1:9030";
```
Comment (yuqi1129) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java:50]: Better change `dorisUri` to `dorisJdbcConnectionUri` and remove the value `http://127.0.0.1:9030` as it's a wrong value, the format should be 'jdbc:mysql:host:post/xxx'

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+  private static final String SCHEMA_TABLE_TITLE = "Tables";
+  private static final String TABLE_TABLE_TITLE = "Columns";
+  private static final String METALAKE_NAME = "test";
+  private static final String CATALOG_TYPE_RELATIONAL = "relational";
+
+  private static final String DORIS_CATALOG_NAME = "catalog_doris";
+  private static final String DORIS_JDBC_DRIVER = "com.mysql.cj.jdbc.Driver";
+  public static final String USER_NAME = "root";
+  public static final String PASSWORD = "root";
+  public static final int FE_MYSQL_PORT = 9030;
```
Comment (yuqi1129) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java:62]: Remove it as it's unused.

**Comment by yuqi1129** [comment]
Please move on for this PR, thanks.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java (hunk 2 above)
Comment (LauraXia123) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java:62]: These are used for property settings of catalog creation

Comment (yuqi1129) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java:62]: image

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+  private static final String CATALOG_TABLE_TITLE = "Schemas";
+  private static final String SCHEMA_TABLE_TITLE = "Tables";
+  private static final String TABLE_TABLE_TITLE = "Columns";
+  private static final String METALAKE_NAME = "test";
+  private static final String CATALOG_TYPE_RELATIONAL = "relational";
+
+  private static final String DORIS_CATALOG_NAME = "catalog_doris";
+  private static final String DORIS_JDBC_DRIVER = "com.mysql.cj.jdbc.Driver";
+  public static final String USER_NAME = "root";
```
Comment (yuqi1129) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java:60]: All fields here can be `private`, no need to make them `public`

Comment (LauraXia123) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java]: fixed

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java (hunk 2 above)
Comment (LauraXia123) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java]: removed

File: build.gradle.kts [hunk 4]
```diff
@@ -154,6 +154,7 @@ allprojects {
       param.environment("HADOOP_USER_NAME", "datastrato")
       param.environment("HADOOP_HOME", "/tmp")
       param.environment("PROJECT_VERSION", project.version)
+      // param.environment("DISPLAY_WEBPAGE_IN_TESTING", true)
```
Comment (yuqi1129) [inline build.gradle.kts:157]: Please remove DISCLAIMER WEBPAGEIN TESTING directly if it is no longer in use.

**Comment by yuqi1129** [comment]
@xunliu 
Please help to take a review if you have time, thanks.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java (hunk 1 above)
Comment (LauraXia123) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageDorisTest.java]: done

File: build.gradle.kts (hunk 4 above)
Comment (LauraXia123) [reply build.gradle.kts]: removed

**Review by xunliu — APPROVED** [review]
When I setting 8GB memory in the Docker, then doris web e2e failed.
When I setting 16GB memory in the Docker, then doris web e2e success.
@LauraXia123 I think you need create a issue to track this problem.

I think we can merged this PR first.

**Comment by xunliu** [comment]
@LauraXia123 
You can setting 16GB memory in the Docker, if CI running in you local success, then we can merge this PR.

**Comment by LauraXia123** [comment]
I can run it success this morning
image

**Comment by xunliu** [comment]
@LauraXia123 Thank you for your contributions.
LGTM

**Comment by jerryshao** [comment]
Again @xunliu , please follow the steps in `MAINTAINERS.md` when merging the PR. I really don't want to say this again and again. Merging a PR is very simple, but it will cost time when doing a release without following the steps. You have to a) weigh the necessity to backport; b) add necessary tags.

**Pull request merged**